src/plugins/unified_search/public/filter_bar/filter_editor/filter_editor.tsx

```diff
@@ -626,6 +628,26 @@ class FilterEditorComponent extends Component<FilterEditorProps, State> {
     this.props.onLocalFilterUpdate?.(newFilter);
   };
 
+  /**
+   * Helper to get unique data views by spec
+   * There can be multiple ES|QL data view with same spec but different id
+   * Those get filtered out because they don't matter in the filter context
+   * Other data views won't get filtered out
+   * @param dataViews[]
+   */
+  private getUniqueSpecHashDataViews(dataViews: DataView[]) {
+    const map = new Map<string, DataView>();
+
+    for (const dataView of dataViews) {
+      const hash = dataView.getSpecHash?.() || dataView.id;
```

Comment:
why can't we rely on having this new method? it seems in alerting, DataView only pretend to be DataViews, and hide their real nature by type casting, something we should fix seperately